master/internal/api_user.go

```diff
@@ -683,3 +696,253 @@ func (a *apiServer) PostUserActivity(
+	if sortColumn != "id" {
+		query.OrderExpr("id asc")
+	}
+
+	err := query.Scan(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = user.AuthZProvider.Get().CanGetAllLongLivedTokens(ctx, *curUser); err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	res := &apiv1.GetAllLongLivedTokensResponse{}
+	for _, s := range userSessions {
+		res.LongLivedTokenInfo = append(res.LongLivedTokenInfo, s.Proto())
+	}
+	return res, api.Paginate(&res.Pagination, &res.LongLivedTokenInfo, req.Offset, req.Limit)
+}
+
+// GET User Long lived token takes user id and returns token information for the given user.
+func (a *apiServer) GetUserLongLivedToken(
+	ctx context.Context, req *apiv1.GetUserLongLivedTokenRequest,
+) (*apiv1.GetUserLongLivedTokenResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	targetFullUser, err := getFullModelUser(ctx, model.UserID(req.UserId))
+	if err != nil {
+		return nil, err
+	}
+	targetUser := targetFullUser.ToUser()
+	if err = user.AuthZProvider.Get().CanGetUsersToken(ctx, *curUser, targetUser); err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	tokenInfo, err := user.GetLongLivedTokenInfo(ctx, targetFullUser.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	return &apiv1.GetUserLongLivedTokenResponse{LongLivedTokenInfo: tokenInfo.Proto()}, nil
+}
+
+// DELETE Long lived token deletes long lived token with current logged in user_id.
+func (a *apiServer) DeleteLongLivedToken(
+	ctx context.Context, req *apiv1.DeleteLongLivedTokenRequest,
+) (*apiv1.DeleteLongLivedTokenResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if err = user.AuthZProvider.Get().CanDeleteUsersOwnToken(ctx, *curUser); err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	err = user.DeleteLongLivenTokenByUserID(ctx, curUser.ID)
+	if err != nil {
+		return nil, err
+	}
+	return &apiv1.DeleteLongLivedTokenResponse{}, nil
+}
+
+// DELETE Long lived token deletes long lived token with given id.
+func (a *apiServer) DeleteLongLivedTokenByTokenID(
+	ctx context.Context, req *apiv1.DeleteLongLivedTokenByTokenIDRequest,
+) (*apiv1.DeleteLongLivedTokenByTokenIDResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	userIDFromTokenID, err := getUserIDFromTokenID(ctx, req.TokenId)
+	if err != nil || userIDFromTokenID < 1 {
+		return nil, err
+	}
+
+	targetFullUser, err := getFullModelUser(ctx, userIDFromTokenID)
+	if err != nil {
+		return nil, err
+	}
+	targetUser := targetFullUser.ToUser()
+	if err = user.AuthZProvider.Get().CanDeleteUsersToken(ctx, *curUser, targetUser); err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	err = user.DeleteSessionByID(ctx, model.SessionID(req.TokenId))
```

Comment:
Missed to update the Delete method here.
It should be -

`err = user.DeleteLongLivenTokenByTokenID(ctx, model.SessionID(req.TokenId))
`